### PR TITLE
check data in fillService facility

### PR DESCRIPTION
**Pull request by AlvaroVega**
time=2021-05-04T11:25:36.533Z | lvl=FATAL | corr=n/a | trans=n/a | op=IoTAgentNG
SI.Global | from=n/a | srv=n/a | subsrv=n/a | msg=An unexpected exception has be
en raised. Ignoring: TypeError: Cannot read property 'service' of undefined
    at fillService (/opt/iotaul/node_modules/iotagent-node-lib/lib/services/comm
on/domain.js:198:14)
    at saveHandler (/opt/iotaul/node_modules/iotagent-node-lib/lib/services/devi
ces/deviceRegistryMongoDB.js:72:26)
    at /opt/iotaul/node_modules/mongoose/lib/model.js:4598:16
    at /opt/iotaul/node_modules/mongoose/lib/utils.js:256:11
    at /opt/iotaul/node_modules/mongoose/lib/model.js:468:16
    at /opt/iotaul/node_modules/kareem/index.js:246:48
    at next (/opt/iotaul/node_modules/kareem/index.js:167:27)
    at next (/opt/iotaul/node_modules/kareem/index.js:169:9)
    at Kareem.execPost (/opt/iotaul/node_modules/kareem/index.js:217:3)
    at _handleWrapError (/opt/iotaul/node_modules/kareem/index.js:245:21)
    at _cb (/opt/iotaul/node_modules/kareem/index.js:304:16)
    at /opt/iotaul/node_modules/mongoose/lib/model.js:324:9
    at /opt/iotaul/node_modules/kareem/index.js:135:16
    at processTicksAndRejections (internal/process/task_queues.js:79:11) | comp=
IoTAgent

IMHO is happen since we updated to nodejs 12 fro node 10

### CODE REVIEW

File: lib/services/common/domain.js [hunk 1]
```diff
@@ -195,11 +195,11 @@ function finishSouthboundTransaction(callback) {
  * @return {Object}             New context containing service information.
  */
 function fillService(context, data) {
-    if (data.service) {
+    if (data && data.service) {
```
Comment (fgalan) [inline lib/services/common/domain.js:198]: CNR entry missing? Or is a trivial fix which doesn't need it?

Comment (AlvaroVega) [reply lib/services/common/domain.js:198]: CNR already updated

Comment (fgalan) [reply lib/services/common/domain.js:198]: Fixed in 5c49c75

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**